### ISSUE TITLE
VBox fails to build on current nixos-unstable-small

### ISSUE THREAD

**Issue by sjau** [NixOS/nixpkgs#31792]
## Issue description

VBox fails to build on current nixos-unstable-small

```
ltLinuxCanForwardAsGso':
SO_UDP' undeclared (first use in this function)
                                                     ^
/nix/store/4m6bmc9k4j8pxli98c41hdwwi8c4k0d3-virtualbox-5.2.0-modsrc/vboxnetflt/include/iprt/cdefs.h:1631:53: note: in definition of m #  define RT_UNLIKELY(expr)     __builtin_expect(!!(expr), 0)
                                                     ^~~~
/tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.c: In function 'vboxNetF/tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.c:1526:47: error: 'SKB_G
  For input:

    '${_qt5_install_prefix}/Qt5@module@/Qt5@module@Config.cmake'

  the old evaluation rules produce:

    '/nix/store/3sn8anz4drz31xl8brzl8q75amdcsyjv-qtbase-5.9.1-dev/lib/cmake/Qt5/Qt5Config.cmake'

  but the new evaluation rules produce:

    '/nix/store/3sn8anz4drz31xl8brzl8q75amdcsyjv-qtbase-5.9.1-dev/lib/cmake/Qt5@module@/Qt5@module@Config.cmake'

  Using the old result for compatibility since the policy is not set.
Call Stack (most recent call first):
  /nix/store/3sn8anz4drz31xl8brzl8q75amdcsyjv-qtbase-5.9.1-dev/lib/cmake/Qt5/Qt5Config.cmake:25 (include)
  runners/bookmarks/CMakeLists.txt:6 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxdrv/generic/RTSemEventWait-2-ex-generic.o
-- Checking for module 'libqalculate'
--   No package 'libqalculate' found
-- Could NOT find Qalculate (missing: QALCULATE_LIBRARIES)
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxdrv/generic/RTSemEventWaitNoResume-2-ex-generic.o
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxdrv/generic/RTSemEventMultiWait-2-ex-generic.o
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxdrv/generic/RTSemEventMultiWaitNoResume-2-ex-generic.o
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxdrv/generic/RTTimerCreate-generic.o
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxdrv/generic/errvars-generic.o
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxdrv/generic/mppresent-generic.o
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxdrv/generic/uuid-generic.o
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxdrv/VBox/log-vbox.o
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxdrv/common/alloc/heapsimple.o
  LD [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxdrv/vboxdrv.o
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetadp/linux/VBoxNetAdp-linux.o
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetadp/VBoxNetAdp.o
  LD [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetadp/vboxnetadp.o
  CC [M]  /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.o
/tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.c: In function 'vboxNetFltLinuxSkBufFromSG':
/tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.c:741:24: error: 'SKB_GSO_UDP' undeclared (first use in this function)
             fGsoType = SKB_GSO_UDP;
                        ^~~~~~~~~~~
/tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.c:741:24: note: each undeclared identifier is reported only once for each function it appears in
In file included from /nix/store/4m6bmc9k4j8pxli98c41hdwwi8c4k0d3-virtualbox-5.2.0-modsrc/vboxnetflt/include/iprt/types.h:29:0,
                 from /nix/store/4m6bmc9k4j8pxli98c41hdwwi8c4k0d3-virtualbox-5.2.0-modsrc/vboxnetflt/r0drv/linux/the-linux-kernel.h:34,
                 from /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.c:24:
/tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.c: In function 'vboxNetFltLinuxCanForwardAsGso':
/tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.c:1276:53: error: 'SKB_GSO_UDP' undeclared (first use in this function)
     if (RT_UNLIKELY( skb_shinfo(pSkb)->gso_type & ~(SKB_GSO_UDP | SKB_GSO_DODGY | SKB_GSO_TCPV6 | SKB_GSO_TCPV4) ))
                                                     ^
/nix/store/4m6bmc9k4j8pxli98c41hdwwi8c4k0d3-virtualbox-5.2.0-modsrc/vboxnetflt/include/iprt/cdefs.h:1631:53: note: in definition of macro 'RT_UNLIKELY'
 #  define RT_UNLIKELY(expr)     __builtin_expect(!!(expr), 0)
                                                     ^~~~
/tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.c: In function 'vboxNetFltLinuxForwardToIntNetInner':
/tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.c:1526:47: error: 'SKB_GSO_UDP' undeclared (first use in this function)
         if (   (skb_shinfo(pBuf)->gso_type & (SKB_GSO_UDP | SKB_GSO_TCPV6 | SKB_GSO_TCPV4))
                                               ^~~~~~~~~~~
make[4]: *** [/nix/store/4qhvpzfmc2zzf339db7vdv8hcp25k4yw-linux-4.14-dev/lib/modules/4.14.0/source/scripts/Makefile.build:315: /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.o] Error 1
make[3]: *** [/nix/store/4qhvpzfmc2zzf339db7vdv8hcp25k4yw-linux-4.14-dev/lib/modules/4.14.0/source/scripts/Makefile.build:573: /tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc/vboxnetflt] Error 2
make[2]: *** [/nix/store/4qhvpzfmc2zzf339db7vdv8hcp25k4yw-linux-4.14-dev/lib/modules/4.14.0/source/Makefile:1503: _module_/tmp/nix-build-virtualbox-modules-5.2.0-4.14.drv-0/virtualbox-5.2.0-modsrc] Error 2
make[1]: *** [Makefile:146: sub-make] Error 2
make: *** [Makefile:24: __sub-make] Error 2
make: Leaving directory '/nix/store/4qhvpzfmc2zzf339db7vdv8hcp25k4yw-linux-4.14-dev/lib/modules/4.14.0/build'
builder for ‘/nix/store/787i7lg420qcpn4gddsvrb45x07w96j6-virtualbox-modules-5.2.0-4.14.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/4sbrmqaw590pvgf41dn7yyhxqcbsr8sg-kernel-modules.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/zhg86f20pm85ihzsbk5j5wnj9x5s1pmj-nixos-system-subi-18.03pre120728.9ce33dd8fb.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/zhg86f20pm85ihzsbk5j5wnj9x5s1pmj-nixos-system-subi-18.03pre120728.9ce33dd8fb.drv’ failed
```

### Steps to reproduce

Add vbox and extension pack to your config, try to build on unstable-small

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.13.12, NixOS, 18.03pre120270.2d2f5e1c4d (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.15`
 - channels(hyper): `"nixos-18.03pre118346.5f8bbb96f8"`
 - channels(root): `"nixos-18.03pre120728.9ce33dd8fb"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)

`18.03pre120270.2d2f5e1c4d (Impala)`

* Nix version: (run `nix-env --version`)

`nix-env (Nix) 1.11.15`

* Nixpkgs version: (run `nix-instantiate --eval "<nixpkgs>" -A lib.nixpkgsVersion`)

`"18.03pre120728.9ce33dd8fb"`

* Sandboxing enabled: (run `grep build-use-sandbox /etc/nix/nix.conf`)

`build-use-sandbox = false`

**Comment by Mic92** [comment]
already tracked here: https://github.com/NixOS/nixpkgs/issues/31640
